actioncable/lib/assets/javascripts/action_cable/connection.coffee

```diff
@@ -75,7 +79,17 @@ class ActionCable.Connection
     error: ->
       @disconnect()
 
+    dispatchEvent: (identifier, type, message) ->
+      unless identifier == "_ping"
+        event = document.createEvent('Event')
+        event.initEvent("cable:#{type}", true, false)
+        channel = JSON.parse(identifier).channel
+        event.cable = {data: message, identifier: channel}
+        document.dispatchEvent(event)
```

Comment:
Would suggest reorganizing the API to:

```coffee
dispatchEvent: (event, data, channel = null) ->
  event = document.createEvent('Event')
  event.initEvent("cable:#{event}", true, false)
  event.cable = {data, channel}
  document.dispatchEvent(event)
```

We'd have to handle the ping differently though. But it feels off were it is now.